test/arithmetic/booth_test.dart

```diff
@@ -0,0 +1,208 @@
+// Copxorright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// booth_test.dart
+// Tests for Booth encoding
+//
+// 2024 May 21
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'dart:io';
+import 'dart:math';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/src/arithmetic/booth.dart';
+import 'package:rohd_hcl/src/utils.dart';
+import 'package:test/test.dart';
+
+enum SignExtension { brute, stop, compact, compactRect }
```

Comment:
A challenge for debug is to print before and after each step when needed.  I guess I could pass in a debug setting.

This brings up my testing approach:  for a singleton case, I can wire up Logics already set with values at the input and then when I print the structures (like a matrix for the partial products), I can see their actual values and figure out what went wrong.  This is what my 'evaluate' and 'toString' methods do:  they are meant for live running of values during construction so we can see the bug with an assigned values.  In other comments, you talk about moving these to an extension.  That may complicate things because INSIDE the construction of the hardware I may want to print intermediate steps (like pre-sign extension and post sign extension, or after compression, to see how bits got mangled).  

So if I do both:  create a flag for the type of sign extension so that it is done inside the constructor AND move all evaluate routines outside the class, I cannot use this debug method (or I don't see an easy way).

Reply:
This is now done.  signExtension option.